xcell/mappers/mapper_HSC_DR1wl.py

```diff
@@ -0,0 +1,278 @@
+        read_lite, fname_lite = self._check_lite_exists(fname)
+        if read_lite:
+            print(f'Loading lite mask {fname}')
+            self.mask = hp.read_map(fname_lite)
+        else:
+            print(f'Computing bin {self.bn} mask')
+            cat = self.get_catalog()
+            self.mask = get_map_from_points(cat, self.nside,
+                                            w=cat[self.w_name],
+                                            ra_name='ra',
+                                            dec_name='dec')
+            if fname_lite is not None:
+                hp.write_map(fname_lite, self.mask, overwrite=True)
+        return self.mask
+
+    def get_nl_coupled(self):
+        if self.nl_coupled is not None:
+            return self.nl_coupled
+
+        fname = f'HSCDR1wl_{self.bn}_w2s2_ns{self.nside}.fits.gz'
+        read_lite, fname_lite = self._check_lite_exists(fname)
+        if read_lite:
+            print(f'Loading lite w2s2 {fname}')
+            w2s2 = hp.read_map(fname_lite)
+        else:
+            print('Computing noise bias')
+            cat = self.get_catalog()
+            w2s2 = get_map_from_points(cat, self.nside,
+                                       w=(0.5*(cat['e1']**2 + cat['e2']**2) *
+                                          cat[self.w_name]**2),
+                                       ra_name='ra', dec_name='dec')
+            if fname_lite is not None:
+                hp.write_map(fname_lite, w2s2, overwrite=True)
+        N_ell = hp.nside2pixarea(self.nside) * np.sum(w2s2) / self.npix
+        nl = N_ell * np.ones(3*self.nside)
+        nl[:2] = 0  # Ylm = for l < spin
+        self.nl_coupled = np.array([nl, 0*nl, 0*nl, nl])
+        return self.nl_coupled
+
+    def get_nz(self, dz=0):
+        if self.dndz is None:
+            fname = f'HSCDR1wl_{self.bn}_nz.npz'
+            read_lite, fname_lite = self._check_lite_exists(fname)
+            if read_lite:
+                print(f'Loading lite nz {fname_lite}')
+                d = np.load(fname_lite)
+                self.dndz = [d['z'], d['nz']]
+            else:
+                print('Computing nz')
+                cat_cosmos = Table.read(self.config['fname_cosmos'])
+                cat_photo = vstack([Table.read(n)
+                                    for n in self.config['fnames_cosmos_ph']])
+                oid, id_ph, id_cs = np.intersect1d(cat_photo['ID'],
+                                                   cat_cosmos['S17a_objid'],
+                                                   return_indices=True)
+                cat_photo = cat_photo[id_ph]
+                cat_cosmos = cat_cosmos[id_cs]
+
+                z0, zf = self.z_edges
+                msk = np.where((cat_photo['PHOTOZ_BEST'] <= zf) &
+                               (cat_photo['PHOTOZ_BEST'] > z0))[0]
+                cosmos_masked = cat_cosmos[msk]
+                w = cosmos_masked['SOM_weight']*cosmos_masked['weight_source']
```

Comment:
Could you add a comment here saying that since this is shear, you have to multiply the shear weights to the DIR weights?

Reply:
done